### PR TITLE
fs: Add common automount procedure for arch independent file systems 

### PR DESCRIPTION
**Pull request by lokeshbv**
- `arch/arm: Remove auto mountng of procfs`
Removes auto mounting of arch independent PROCFS from artik05x & sidk_s5jt200 tash initialization routines.

- `fs: support auto mounting of arch independent file systems`
Adds support for auto mounting of architecture independent File Systems like procfs, tmpfs, etc., at the end of OS side fs initalization.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d48ce08df894c3d778d5619d2aeef50d155e45b2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d48ce08df894c3d778d5619d2aeef50d155e45b2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [93687b668e71db7e453708b1f377def85378f83b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [93687b668e71db7e453708b1f377def85378f83b] - Code Rule Check OK.

File: os/fs/fs_initialize.c [hunk 1]
```diff
@@ -82,6 +84,33 @@
  * Private Functions
  ****************************************************************************/
 
+static void fs_common_automount(void)
```
Comment (Taejun-Kwon) [inline os/fs/fs_initialize.c]: Actually we initiate fs_initialize only for inode for now, In my opinion mount should be called after all peripheral initialized.
procfs including smartfs and something else, I think all other file's changes are acceptable, but not this one.

File: os/fs/procfs/Kconfig [hunk 2]
```diff
@@ -14,6 +14,14 @@ config FS_PROCFS
 		mount at /proc.  Then information about all of the currently active
 		tasks and threads will be available in /proc.
 
+config FS_AUTOMOUNT_PROCFS
```
Comment (jeongarmy) [inline os/fs/procfs/Kconfig]: Please move down in 'if FS_PROCFS'.

**Comment by seinfra** [comment]
Target : [7b81eb3d2b7f1633f523630f334ae508cb70c429] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7b81eb3d2b7f1633f523630f334ae508cb70c429] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5feafac8a4fa35f469be65f199771165bc7ec790] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5feafac8a4fa35f469be65f199771165bc7ec790] - Code Rule Check OK.

**Comment by lokeshbv** [comment]
@Taejun-Kwon please check the latest changes

**Comment by seinfra** [comment]
Target : [c9d57db21939e2dee2c5a6e5d87d62b02093c022] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c9d57db21939e2dee2c5a6e5d87d62b02093c022] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [759c2728f994cdd82eefb04af308e9eaea13eed6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [759c2728f994cdd82eefb04af308e9eaea13eed6] - Code Rule Check (C++) OK.

File: os/fs/procfs/Kconfig [hunk 3]
```diff
@@ -16,6 +16,14 @@ config FS_PROCFS
 
 if FS_PROCFS
 
+config FS_AUTOMOUNT_PROCFS
+	bool "Auto mount PROCFS File System"
+	default n
+	depends on FS_PROCFS
```
Comment (jeongarmy) [inline os/fs/procfs/Kconfig]: I think it is not needed because it already is in condition, 'FS_PROCFS'

**Comment by seinfra** [comment]
Target : [0f98e53cd5c6d63962bb0046bf3bc0973847e6c0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0f98e53cd5c6d63962bb0046bf3bc0973847e6c0] - Code Rule Check OK.

**Pull request merged**